corehq/apps/app_manager/static/app_manager/js/widgets_v4.js

```diff
@@ -53,12 +54,33 @@ hqDefine("app_manager/js/widgets_v4", [
             width: options.width || '200px',
         });
 
+        if ($('#app-profile-id-input').length) {
+            $select.on('select2:select', function (e) {
+                var buildProfiles = _.map(e.params.data.buildProfiles, function (details, profileId) {
+                    return { id: profileId, text: details.name };
+                });
+                buildProfiles = Array.prototype.concat({id: 'null', text: gettext("Select profile")},
+                    buildProfiles);
+                $('#app-profile-id-input').select2({data: buildProfiles});
+            });
+            $select.on('change.select2', function () {
+                // https://stackoverflow.com/a/32115793
+                // clear all options manually since it's not getting deleted in this version of select2
+                $('#app-profile-id-input').html('').select2({data: {id: null, text: null }});
+                $('#app-profile-id-input').val(null).trigger('change');
+            });
+        }
+
         if (options.initialValue) {
             // https://select2.org/programmatic-control/add-select-clear-items#preselecting-options-in-an-remotely-sourced-ajax-select2
             var option = new Option(options.initialValue.text, options.initialValue.id, true, true);
             $select.append(option).trigger('change');
             $select.trigger({type: 'select2:select', params: {data: options.initialValue}});
         }
+        var appProfileInitialValue = initialPageData.get('appProfileInitialValue');
+        if (!_.isEmpty(appProfileInitialValue)) {
+            $('#app-profile-id-input').select2({data: appProfileInitialValue});
```

Comment:
Doesn't the input need to be transformed to a select2 regardless of whether or not there's data?

Reply:
If there is no initial value, there is no need to set anything. The select options will populate when the user selects the version. Hence this was skipped.